server.js

```diff
@@ -1279,10 +1285,20 @@ function objectBeatSender(PORT, thisId, thisIp, oneTimeOnly) {
                     });
                 }
             }
-        }, beatInterval + utilities.randomIntInc(-250, 250));
-    } else {
+        };
+
+        // send one beat immediately and then start interval timer triggering beats
+        if (immediate) {
+            sendBeat();
+        }
+        // perturb the inverval a bit so that not all objects send the beat in the same time.
+        activeHeartbeats[thisId] = setInterval(sendBeat, beatInterval + utilities.randomIntInc(-250, 250));
+    } 
+    else {
         // Single-shot, one-time heartbeat
         // delay the signal with timeout so that not all objects send the beat in the same time.
+        let interval = immediate ? 0 : utilities.randomIntInc(1, 250);
```

Comment:
I might rename "interval" here to "delay", otherwise I expect it will repeat multiple times